src/vs_seeker.c

```diff
@@ -0,0 +1,776 @@
+    /*0x431*/ u8 trainerWantsRematch:1;
+    u8 responseCode:5;
+};
+
+// static declarations
+static EWRAM_DATA struct VsSeekerStruct *sVsSeeker = NULL;
+
+static void VsSeekerResetInBagStepCounter(void);
+static void VsSeekerResetChargingStepCounter(void);
+static void Task_ResetObjectsRematchWantedState(u8 taskId);
+static void ResetMovementOfRematchableTrainers(void);
+static void Task_VsSeeker_1(u8 taskId);
+static void Task_VsSeeker_2(u8 taskId);
+static void GatherNearbyTrainerInfo(void);
+static void Task_VsSeeker_3(u8 taskId);
+static bool8 CanUseVsSeeker(void);
+static u8 GetVsSeekerResponseInArea(void);
+static u8 GetRunningBehaviorFromGraphicsId(u8 graphicsId);
+static u16 GetTrainerFlagFromScript(const u8 * script);
+static void ClearAllTrainerRematchStates(void);
+static bool8 IsTrainerVisibleOnScreen(struct VsSeekerTrainerInfo * trainerInfo);
+static u8 GetRematchableTrainerLocalId(void);
+static void StartTrainerObjectMovementScript(struct VsSeekerTrainerInfo * trainerInfo, const u8 * script);
+static u8 GetCurVsSeekerResponse(s32 vsSeekerIdx, u16 trainerIdx);
+static void StartAllRespondantIdleMovements(void);
+static bool8 ObjectEventIdIsSane(u8 objectEventId);
+static u8 GetRandomFaceDirectionMovementType();
+
+static const u8 sMovementScript_Wait48[] = {
+    MOVEMENT_ACTION_DELAY_16,
+    MOVEMENT_ACTION_DELAY_16,
+    MOVEMENT_ACTION_DELAY_16,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sMovementScript_TrainerUnfought[] = {
+    MOVEMENT_ACTION_EMOTE_EXCLAMATION_MARK,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sMovementScript_TrainerNoRematch[] = {
+    MOVEMENT_ACTION_EMOTE_X,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sMovementScript_TrainerRematch[] = {
+    MOVEMENT_ACTION_WALK_IN_PLACE_FASTER_DOWN,
+    MOVEMENT_ACTION_EMOTE_DOUBLE_EXCL_MARK,
+    MOVEMENT_ACTION_STEP_END
+};
+
+static const u8 sFaceDirectionMovementTypeByFacingDirection[] = {
+    MOVEMENT_TYPE_FACE_DOWN,
+    MOVEMENT_TYPE_FACE_DOWN,
+    MOVEMENT_TYPE_FACE_UP,
+    MOVEMENT_TYPE_FACE_LEFT,
+    MOVEMENT_TYPE_FACE_RIGHT
+};
+
+void VsSeekerFreezeObjectsAfterChargeComplete(void)
+{
+    CreateTask(Task_ResetObjectsRematchWantedState, 80);
+}
+
+static void Task_ResetObjectsRematchWantedState(u8 taskId)
+{
+    struct Task *task = &gTasks[taskId];
+    u8 i;
+
+    if (task->data[0] == 0 && IsPlayerStandingStill() == TRUE)
+    {
+        PlayerFreeze();
+        task->data[0] = 1;
+    }
+
+    if (task->data[1] == 0)
+    {
+        for (i = 0; i < OBJECT_EVENTS_COUNT; i++)
+        {
+            if (ObjectEventIdIsSane(i) == TRUE)
+            {
+                if (gObjectEvents[i].singleMovementActive)
+                    return;
+                FreezeObjectEvent(&gObjectEvents[i]);
+            }
+        }
+    }
+
+    task->data[1] = 1;
+    if (task->data[0] != 0)
+    {
+        DestroyTask(taskId);
+        StopPlayerAvatar();
+        ScriptContext_Enable();
+    }
+}
```

Comment:
I think this function can be a lot cleaner, if only by giving sensible names to `data[0]` and `data[1]`.

Reply:
I'll be honest, I can't make heads or tails of what they're supposed to do aside from helping set the sequence (since, if I'm understanding correctly, these will always start with 0). I named them anyways.